### ISSUE TITLE
Invalid "simg4" hostname for Gelbooru

### ISSUE THREAD

**Issue by brazenvoid** [Bionus/imgbrd-grabber#1410]
**Bug description**

Many images get picked up with the host name "https://simg4.gelbooru.com/" which is invalid.

**Steps to reproduce**

1. Search any tag combination
2. Use selection of images in browser and "Get selected" button to gather the download links
3. Download the images with "https://simg4.gelbooru.com/" listed as domain
4. See error in log

**Expected behavior**

Replacing the simg4 with simg3 resolves to correct images. Thus, it should do this from the get go.

**System information**

- OS: Windows 10
- Grabber version: 6.0.5

**Comment by Bionus** [comment]
Indeed, a few years ago `https://gelbooru.com/...` didn't work for images, so when a file url had this, Grabber automatically adds the `simg4` sub-domain.

Since it's no longer necessary, I removed the relevant part. :+1:

**Comment by brazenvoid** [comment]
@Bionus I saw it in the code... It was really baffling to what caused you to add such a line. I had actually thought that the bug was on the Gelbooru's end and you may add a workaround...

Anyways it is still not fixed in nightly. The affected files have certainly reduced in number. Now they do not come up as randomly rather they now are coming in batches. Like ~90 images with correct domain then ~30 with bad and so on.

**Comment by Belmont1123** [comment]
yeah this sounds like the issue im having. i was able to dl 1 tag of images but after that i get a message saying an error occurred saving the image. how do i fix this? its weird i was able to dl atleast 1 tag of images but nothing else.

**Comment by brazenvoid** [comment]
@Belmont1123 Well the fix is a workaround essentially and as far as i know, works only with a batch made through manually selecting images in the browser using the "Get Selected Images" button.

Here is what you will have to do:
1. Browse through the tag combinations you want to download
2. Select images on each page that you want to download with Ctrl+Left Click and then clicking the "Get Selected Images" button.
3. Apply Point 2 till you have browsed through all the searches and associated pages. Better yet make your searches to the point so as to avoid wasting a long time. An extensive blacklist and post filters will also help a lot. If you are like me then you will nonetheless waste the time while filtering after a usual tag batch so its not really that wasteful of a way to make a collection.
4. When you are done, click the "Open options folder" shortcut in the file menu. Close Grabber.
5. Go into the "Grabber" folder. Open the restore.igl file in a notepad app.
6. Do a search and replace action with "simg4" as subject and "simg3" as replacement. 
7. Open Grabber and download.

**Comment by Belmont1123** [comment]
holy cow. alright ill try this out. another question tho. when i attempt the download the files that have an error are labled "simg3.gelbooru" rather than simg4. just wanna make sure im actually having the same issue or if this needs a different solution.

**Comment by brazenvoid** [comment]
@Belmont1123 No I was referring to links with simg4. simg3 is working fine for me,

**Comment by brazenvoid** [comment]
@Belmont1123 You may try with a proxy. 
simg3 is blocked in my country so I use a VPN...

**Comment by Belmont1123** [comment]
is it strange that a previous tag images i download before worked but then attempting to re download the same tag images to a different folder results in the same error?

**Comment by brazenvoid** [comment]
@Belmont1123 First get the error. Though it only mentions an aggregate error message for all images but the actual errors will be shown in the "Log" tab.

**Comment by Bionus** [comment]
> @Bionus I saw it in the code... It was really baffling to what caused you to add such a line. I had actually thought that the bug was on the Gelbooru's end and you may add a workaround...

This line is many years old, and was necessary at the time since Gelbooru did not allow to load images from their `https://gelbooru.com` endpoint but only via their static image servers.

> Anyways it is still not fixed in nightly. The affected files have certainly reduced in number. Now they do not come up as randomly rather they now are coming in batches. Like ~90 images with correct domain then ~30 with bad and so on.

I just tried downloading 10,000 images from Gelbooru, none of them used `simg4`. Was the Gelbooru model file properly updated? i.e `C:\Users\Nicolas\AppData\Local\%USERNAME%\Grabber\sites\Gelbooru (0.2)\model.js` should **not** contain:
```javascript
img["file_url"] = img["file_url"].replace("/gelbooru.com/", "/simg4.gelbooru.com/");
```
around the 7th line.

Otherwise, do you have an example of search that gives that many failing images?

**Comment by brazenvoid** [comment]
Yes, the latest nightly is working fine!

**Comment by Neito-kun** [comment]
I know this thread is already closed, but I'd like to share my personal experience with this issue and a _strange way_ to "fix it": 

 I'm currently using Grabber 6.0.5, my system is Windows 7 x64 Professional. I recently started to download a small gallery from an artist using Danbooru, Gelbooru and Hijiribe as sources, with default settings (no login credentials). And I ended up running into this issue (_simg4.gelbooru.com not found_) right after Grabber downloaded all the images available in Danbooru's servers first without any problems... So, I decided to press **skip** button and then I **closed** the download dialog box (cancelling with it all the attempts to download the rest of the files from **simg4.gelbooru.com**...). After that, I pressed again the **Download** button, and Grabber, once again, started the job, skipping the already downloaded images from Danbooru, but then, when Gelbooru's images' urls started to load, they were recognized this time as **simg3** and Grabber started downloading them without any issues.

 I'm not sure if this solution is possible for every case, but I share it here anyway, in case someone else could replicate the same behavior with any images gallery. 

 As a foot note, I keep having troubles to download big galleries (over ~1000 pages) but I think it's related to Danbooru's servers or databases...
